systems/trajectory_optimization/hybrid_dircon.cc

```diff
@@ -418,6 +419,129 @@ void HybridDircon<T>::SetInitialForceTrajectory(
   SetInitialGuess(collocation_slack_vars_[mode], guess_collocation_slack);
 }
 
+template <typename T>
+void HybridDircon<T>::ScaleTimeVariables(double scale) {
+  for (int i = 0; i < h_vars().size(); i++) {
+    DRAKE_DEMAND(IsVariableScalingUnset(h_vars()(i)));
```

Comment:
What's wrong with re-writing the scale factor? Why is this DEMAND necessary?

Reply:
I used DEMAND to avoid unintentional rewriting. I guess I can remove it so that it's not so restrictive

Reply:
I think we can trust users to rewrite only if they mean it.

Reply:
I removed `DRAKE_DEMAND`. It's easy for users to check if a variable scaling has been set